src/Elastic.Apm/Logging/LogValuesFormatter.cs

```diff
@@ -46,19 +49,41 @@ public LogValuesFormatter(string format, string scope = null)
 					// Format item syntax : { index[,alignment][ :formatString] }.
 					var formatDelimiterIndex = FindIndexOfAny(format, FormatDelimiters, openBraceIndex, closeBraceIndex);
 
-					sb.Append(format, scanIndex, openBraceIndex - scanIndex + 1);
-					sb.Append((ValueNames.Count).ToString(CultureInfo.InvariantCulture));
-					ValueNames.Add(format.Substring(openBraceIndex + 1, formatDelimiterIndex - openBraceIndex - 1));
-					sb.Append(format, formatDelimiterIndex, closeBraceIndex - formatDelimiterIndex + 1);
+					if (ValueNames.Count < expectedArgs)
+					{
+						sb.Append(format, scanIndex, openBraceIndex - scanIndex + 1);
+						sb.Append(ValueNames.Count.ToString(CultureInfo.InvariantCulture));
+						ValueNames.Add(format.Substring(openBraceIndex + 1, formatDelimiterIndex - openBraceIndex - 1));
+						sb.Append(format, formatDelimiterIndex, closeBraceIndex - formatDelimiterIndex + 1);
+					}
+					else
+					{
+						_isArgumentNumberMismatch = true;
+						_mismatchedValueNames.Add(format.Substring(openBraceIndex + 1, formatDelimiterIndex - openBraceIndex - 1));
+					}
 
 					scanIndex = closeBraceIndex + 1;
 				}
 			}
 
+			if (_isArgumentNumberMismatch)
+			{
+				sb.AppendLine();
+				sb.AppendLine(
+					$"Above line is from an invalid structured log and may not be complete: number of arguments is not matching the number of placeholders, placeholders with missing values: {string.Join(", ", _mismatchedValueNames.ToArray())}");
```

Comment:
As I mentioned before, having it on the same line as the message itself will make life easier for those of us that use grep and such.

Reply:
Fine by me - now it's in the same line.